### PR TITLE
session_0 and session_1 arshad

### PR DESCRIPTION
**Pull request by arshadace**

### CODE REVIEW

File: session_0/fizz_buzz.rb [hunk 1]
```diff
@@ -37,5 +37,21 @@
 # We will raise an `ArgumentError` exception to let the caller know that
 # their function arguments were incorrect.
 def fizz_buzz(n:, x: 3, y: 5)
-  raise NotImplementedError # TODO
+  raise ArgumentError if n<0||x<=0||y<=0
```
Comment (abhishekkumar2718) [inline session_0/fizz_buzz.rb:40]: https://rubystyle.guide/#spaces-operators

File: session_0/fizz_buzz.rb [hunk 2]
```diff
@@ -37,5 +37,21 @@
 # We will raise an `ArgumentError` exception to let the caller know that
 # their function arguments were incorrect.
 def fizz_buzz(n:, x: 3, y: 5)
-  raise NotImplementedError # TODO
+  raise ArgumentError if n<0||x<=0||y<=0 
+  
+  l=[]
+  for i in 1..n
```
Comment (abhishekkumar2718) [inline session_0/fizz_buzz.rb:43]: Can map function be used instead?

https://www.rubyguides.com/2018/10/ruby-map-method/

File: session_0/word_processing.rb [hunk 3]
```diff
@@ -2,13 +2,16 @@
...
+    
+  return words.map {|w| w.downcase}
 end
 
 # Similar to `lower_case`, this function modifies the array in-place
 # and does not return any value.
 def lower_case!(words)
-  raise NotImplementedError # TODO
+  
+  puts words.map! {|w| w.downcase}
```
Comment (abhishekkumar2718) [inline session_0/word_processing.rb:13]: You don't need to print words to the console

File: session_0/word_processing.rb [hunk 4]
```diff
@@ -34,11 +38,31 @@ def words_with_prefix(prefix, words)
...
-  raise NotImplementedError # TODO
+  d=0
+  for i in 1..(word_1.length<word_2.length ? word_1.length: word_2.length)
+    if (word_1[i-1]==word_2[i-1])
+      d=d+1  
+    else
+      break
+    end
+  end  
+    return d
```
Comment (abhishekkumar2718) [inline session_0/word_processing.rb:49]: Misleading indentation

File: session_0/word_processing.rb [hunk 5]
```diff
@@ -34,11 +38,31 @@ def words_with_prefix(prefix, words)
...
 end
 
 # Given a chosen word and an array of words, return an array of word(s)
 # with the maximum similarity score in the order they appear.
+
 def most_similar_words(chosen_word, words)
-  raise NotImplementedError # TODO
+  d=[]
+  s=[]
+  for i in 1..words.length
```
Comment (abhishekkumar2718) [inline session_0/word_processing.rb:58]: Use map function instead

**Review by abhishekkumar2718 — CHANGES_REQUESTED** [review]

File: session_0/word_processing.rb (hunk 3 above)
Comment (addhyanmalhotra) [inline session_0/word_processing.rb:13]: the 'puts' is unnecessary

File: session_0/word_processing.rb [hunk 6]
```diff
@@ -34,11 +38,31 @@ def words_with_prefix(prefix, words)
 # The function `similarity_score` takes two words and returns the
 # similarity score (an integer).
 def similarity_score(word_1, word_2)
-  raise NotImplementedError # TODO
+  d=0
+  for i in 1..(word_1.length<word_2.length ? word_1.length: word_2.length)
+    if (word_1[i-1]==word_2[i-1])
```
Comment (addhyanmalhotra) [inline session_0/word_processing.rb:43]: try using slices, it improves readability of the code
https://apidock.com/ruby/String/slice

You can also avoid the ternary operator by using max method
https://stackoverflow.com/questions/1359370/how-to-find-a-min-max-with-ruby

**Pull request closed without merging**